back/api/views.py

```diff
@@ -72,6 +83,38 @@ def post(self, request):
         }
         return response
 
+class TeacherLoginView(APIView):
+    def post(self, request):
+        username = request.data['username']
+        password = request.data['password']
+
+
+        user = User.objects.get(username=username)
+        teacher = Teacher.objects.get(user=user.id)
```

Comment:
why not simply access the user throught the teacher directly ?

Reply:
Indeed, I simplified the two lines in one